### PR TITLE
feat: delegation.toJSON

### LINKED ISSUE #172: Add toJSON method to delegations

### PR DESCRIPTION
**Pull request by Gozala**
This supersedes #173 and Fixes #172. It targets #185 as it relies on changes there.

In the other PR @alanshaw had this suggestion

> If we get a fromJSON we can remove some code for storing delegations in the access Conf store.

And this PR still does not add `fromJSON` because, even though it will serialize the whole chain into a JSON which could be then de-serialized into compatible chain there is still a problem with the fact that `fromJSON` would not know if inline proofs do match claimed CIDs. This means that motivated attacker could use this an a attack vector.

In other words code that imports delegation should also validate hashes which is async operation and there for does not fit into a `fromJSON` signature.

That said I think it still useful to have `toJSON` method for debugging purposes, which is what this PR adds.

### CODE REVIEW

**Review by gobengo — APPROVED** [review]
awesome

**Review by alanshaw — APPROVED** [review]

**Review by vasco-santos — APPROVED** [review]
lgtm thanks @Gozala
